apps/admin/backend/src/util/manual_tallies.ts

```diff
@@ -0,0 +1,339 @@
+          tally: 0,
+        };
+        break;
+      }
+      default:
+        throwIllegalValue(contest, 'type');
+    }
+    contestTallies[contest.id] = {
+      contest,
+      tallies: optionTallies,
+      metadata: { overvotes: 0, undervotes: 0, ballots: 0 },
+    };
+  }
+  return contestTallies;
+}
+
+/**
+ * @deprecated Copied from `admin/frontend` to use during manual tally data
+ * transition.
+ */
+function getEmptyManualTalliesByPrecinct(
+  election: Election
+): Dictionary<ManualTally> {
+  const tallies: Dictionary<ManualTally> = {};
+  for (const precinct of election.precincts) {
+    tallies[precinct.id] = {
+      contestTallies: getEmptyContestTallies(election),
+      numberOfBallotsCounted: 0,
+    };
+  }
+  return tallies;
+}
+
+/**
+ * @deprecated Copied from `admin/frontend` to use during manual tally data
+ * transition. We use this to create a {@link FullElectionManualTally} which
+ * will no longer be served to the frontend once tally reports are created on
+ * the backend.
+ */
+function convertTalliesByPrecinctToFullManualTally(
+  talliesByPrecinct: Dictionary<ManualTally>,
+  election: Election,
+  votingMethod: VotingMethod,
+  timestampCreated: Date
+): FullElectionManualTally {
+  let totalNumberOfBallots = 0;
+  const overallContestTallies: Dictionary<ContestTally> = {};
+  for (const precinctTally of Object.values(talliesByPrecinct)) {
+    assert(precinctTally);
+    totalNumberOfBallots += precinctTally.numberOfBallotsCounted;
+    for (const contestId of Object.keys(precinctTally.contestTallies)) {
+      if (!(contestId in overallContestTallies)) {
+        overallContestTallies[contestId] =
+          precinctTally.contestTallies[contestId];
+      } else {
+        const existingContestTallies = overallContestTallies[contestId];
+        const secondTally = precinctTally.contestTallies[contestId];
+        assert(existingContestTallies);
+        assert(secondTally);
+        overallContestTallies[contestId] = combineContestTallies(
+          existingContestTallies,
+          secondTally
+        );
+      }
+    }
+  }
+
+  const overallTally: ManualTally = {
+    contestTallies: overallContestTallies,
+    numberOfBallotsCounted: totalNumberOfBallots,
+  };
+
+  const resultsByCategory = new Map();
+  resultsByCategory.set(TallyCategory.Precinct, talliesByPrecinct);
+
+  // Compute results filtered by party, this filters the sets of contests and requires recomputing the number of ballots counted.
+  const contestTalliesByParty: Dictionary<ManualTally> = {};
+  const partiesInElection = getPartiesWithPrimaryElections(election);
+  for (const party of partiesInElection) {
+    contestTalliesByParty[party.id] = filterTallyForPartyId(
+      overallTally,
+      party.id,
+      election
+    );
+  }
+  resultsByCategory.set(TallyCategory.Party, contestTalliesByParty);
+
+  return {
+    overallTally,
+    resultsByCategory,
+    votingMethod,
+    timestampCreated,
+  };
+}
```

Comment:
This is all just copied from the frontend. It will someday be removed from the frontend. I could have pulled it out into `libs/utils` but was easiest just to copy, and it will be deleted from the frontend later on in this project.